src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -429,10 +429,19 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 		PlanState *planState = &(scanState->customScanState.ss.ps);
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
+		ExecuteTaskListFunctions(taskList, planState);
```

Comment:
I guess? But isn't that, uh, planning? The lines between planning and execution are getting real blurry here. The planning phase (when possible) prunes and groups all rows by their destination shards. Those groups become tasks, whose regrouped rows we use to generate the final SQL.

What you suggest I could see, but it would drive a wedge between what we do in planning and what we do in execution (i.e. if fully plannable ahead of time, we'd create these tasks like I do now, but if not, we'd build things directly from the Query at execution).